src/GraphQL.Tests/Types/CustomScalarTests.cs

```diff
@@ -68,7 +68,7 @@ public void ToAST(string internalValue, string externalValue)
         [InlineData("inputOutput", null, null, Response.Success)]                     // custom scalar coerces null to "internalNull" and then custom scalar coerces "internalNull" to null
         [InlineData("inputOutput", "internalNull", null, Response.Success)]           // custom scalar receives "internalNull" and then custom scalar coerces "internalNull" to null
         [InlineData("inputOutput", "externalNull", "externalNull", Response.Success)] // custom scalar coerces "externalNull" to null and then custom scalar coerces null to "externalNull"
-        [InlineData("nonNullOutput", "internalNull", null, Response.ErrorNoField)]    // string scalar receives "internalNull" and non-null custom scalar attempts to return null (and fails)
+        [InlineData("nonNullOutput", "internalNull", null, Response.ErrorNoData)]     // string scalar receives "internalNull" and non-null custom scalar attempts to return null (and fails)
```

Comment:
I'm puzzled here.  From what it looks like, this was incorrect before.  Specifically, the null value should propagate up because the non-null field attempted to return null.  But there was almost no changes to the serialization process -- only deserialization was added.  I'm sure we have tests for null propagation.  In fact another test in this same set does indeed propagate null properly.  So I will need to check this out.  Ditto with the other change in this file.

Reply:
Figured it out.  According to GraphQL spec, a document that fails validation must not contain `data` in the response map, whereas a document that fails during execution must return `"data": null`.  So it was correct before.  Had to make another change though, that not sure why I had to make.  But at least it's operating correctly.